src/canmatrix/canmatrix.py

```diff
@@ -870,8 +870,8 @@ def decode(self, data, decodeToStr=False):
                     signals = sorted(self.signals, key=lambda s: s.getStartbit())
                     signals_values = OrderedDict()
                     for signal, value in zip(signals, bitstruct.unpack(fmt, data)):
-                        signals_values[signal.name] = signal.raw2phys(value, decodeToStr)
-                muxVal = int(signals_values.values()[0])
+                        signals_values[signal.name] = signal.raw2phys(defaultFloatFactory(value), decodeToStr)
```

Comment:
What wasn't working that you added this?  Perhaps add a test to `src/canmatrix/tests/test_canmatrix.py` that was broken and gets fixed by the change.

Reply:
I added a test with more explanation in the follow up PR #215.